core/src/main/java/org/pentaho/di/connections/vfs/provider/ConnectionFileObject.java

```diff
@@ -22,102 +22,144 @@
 
 package org.pentaho.di.connections.vfs.provider;
 
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
 import org.apache.commons.vfs2.FileContent;
 import org.apache.commons.vfs2.FileObject;
 import org.apache.commons.vfs2.FileSelector;
 import org.apache.commons.vfs2.FileSystemException;
 import org.apache.commons.vfs2.FileType;
 import org.apache.commons.vfs2.RandomAccessContent;
 import org.apache.commons.vfs2.operations.FileOperations;
-import org.apache.commons.vfs2.provider.AbstractFileName;
 import org.apache.commons.vfs2.provider.AbstractFileObject;
-import org.apache.commons.vfs2.provider.URLFileName;
 import org.apache.commons.vfs2.util.RandomAccessMode;
 import org.pentaho.di.core.vfs.AliasedFileObject;
-import org.pentaho.di.core.vfs.KettleVFS;
 
 import java.io.InputStream;
 import java.io.OutputStream;
 
 /**
  * The FileObject implementation for the PVFS (Pentaho VFS) vfs provider.
- *
+ * <p>
  * The purpose of ConnectionFileObject is to represent a named connection FileObject. ConnectionFileObject holds a
  * reference to the resolved FileObject and delegates metadata calls to that reference object, but for other calls, such
  * as getChildren, ConnectionFileObject resolves the child FileObjects, but converts them to pvfs FileObjects to
  * maintain named connection information.
  */
+@SuppressWarnings( "resource" )
 public class ConnectionFileObject extends AbstractFileObject<ConnectionFileSystem> implements AliasedFileObject {
 
   public static final String DELIMITER = "/";
-  private final AbstractFileObject<ConnectionFileSystem> resolvedFileObject;
-  private final String domain;
+  public static final char DELIMITER_CHAR = DELIMITER.charAt( 0 );
+
+  @Nullable
+  private final AbstractFileObject<?> resolvedFileObject;
```

Comment:
In part, the changes in this file address https://hv-eng.atlassian.net/browse/BACKLOG-40732.
Also, the comments and changes clarify the cases in which `ConnectionFileObject` and `ConnectionFileName` can be used.
Also, removed the `domain` field and `getChild(.)` --- the main changes on this file. The file object now calls back on its file system --- a field it already has --- to do this: `ConnectionFileSystem#createChild(.)`.